File: crates/pulse-infrastructure/src/flash_api/queries.rs
//! Flash API GraphQL queries

/// Query to get user info and wallets
pub const ME_WALLETS_QUERY: &str = r#"
  query meWallets {
    me {
      id
      phone
      username
      defaultAccount {
        id
        defaultWalletId
        wallets {
          id
          balance
          walletCurrency
        }
        realtimePrice {
          id
          timestamp
          denominatorCurrency
          btcSatPrice {
            base
            offset
          }
          usdCentPrice {
            base
            offset
          }
        }
      }
    }
  }
"#;

/// Query to get real-time BTC price
pub const REALTIME_PRICE_QUERY: &str = r#"
  query realtimePrice($currency: DisplayCurrency = "USD") {
    realtimePrice(currency: $currency) {
      id
      timestamp
      denominatorCurrency
      btcSatPrice {
        base
        offset
      }
      usdCentPrice {
        base
        offset
      }
    }
  }
"#;

/// Query to get transaction history for default account
pub const TRANSACTION_LIST_QUERY: &str = r#"
  query transactionListForDefaultAccount(
    $first: Int
    $after: String
  ) {
    me {
      id
      defaultAccount {
        id
        transactions(first: $first, after: $after) {
          pageInfo {
            hasNextPage
            hasPreviousPage
            startCursor
            endCursor
          }
          edges {
            cursor
            node {
              id
              status
              direction
              memo
              createdAt
              settlementAmount
              settlementCurrency
            }
          }
        }
      }
    }
  }
"#;

/// Query to check if username is available
pub const USERNAME_AVAILABLE_QUERY: &str = r#"
  query usernameAvailable($username: Username!) {
    usernameAvailable(username: $username)
  }
"#;

/// Mutation to create captcha challenge
pub const CAPTCHA_CREATE_CHALLENGE_MUTATION: &str = r#"
  mutation captchaCreateChallenge {
    captchaCreateChallenge {
      errors {
        message
      }
      result {
        id
        challengeCode
        newCaptcha
        failbackMode
      }
    }
  }
"#;

/// Mutation to request auth code (OTP)
pub const CAPTCHA_REQUEST_AUTH_CODE_MUTATION: &str = r#"
  mutation captchaRequestAuthCode($input: CaptchaRequestAuthCodeInput!) {
    captchaRequestAuthCode(input: $input) {
      errors {
        message
      }
      success
    }
  }
"#;

/// Mutation to login with phone and OTP code
pub const USER_LOGIN_MUTATION: &str = r#"
  mutation userLogin($input: UserLoginInput!) {
    userLogin(input: $input) {
      errors {
        message
      }
      authToken
    }
  }
"#;

File: src/modules/wallet/graphql/queries.ts
export const TRANSACTION_LIST_QUERY = `
  query transactionListForDefaultAccount(
    $first: Int
    $after: String
    $last: Int
    $before: String
  ) {
    me {
      id
      defaultAccount {
        id
        transactions(first: $first, after: $after, last: $last, before: $before) {
          pageInfo {
            hasNextPage
            hasPreviousPage
            startCursor
            endCursor
          }
          edges {
            cursor
            node {
              __typename
              id
              status
              direction
              memo
              createdAt
              settlementAmount
              settlementFee
              settlementDisplayFee
              settlementCurrency
              settlementDisplayAmount
              settlementDisplayCurrency
              settlementPrice {
                base
                offset
                currencyUnit
                formattedAmount
              }
              initiationVia {
                ... on InitiationViaIntraLedger {
                  counterPartyWalletId
                  counterPartyUsername
                }
                ... on InitiationViaLn {
                  paymentHash
                }
                ... on InitiationViaOnChain {
                  address
                }
              }
              settlementVia {
                ... on SettlementViaIntraLedger {
                  counterPartyWalletId
                  counterPartyUsername
                }
                ... on SettlementViaLn {
                  paymentSecret
                }
                ... on SettlementViaOnChain {
                  transactionHash
                }
              }
            }
          }
        }
      }
    }
  }
`;

export const REALTIME_PRICE_QUERY = `
  query realtimePrice {
    me {
      defaultAccount {
        realtimePrice {
          btcSatPrice {
            base
            offset
          }
          usdCentPrice {
            base
            offset
          }
          denominatorCurrency
        }
      }
    }
  }
`;

export const ACCOUNT_DEFAULT_WALLET_QUERY = `
  query accountDefaultWallet($username: Username!) {
    accountDefaultWallet(username: $username) {
      id
    }
  }
`;

export const ME_WALLETS_QUERY = `
  query meWallets {
    me {
      id
      defaultAccount {
        id
        defaultWalletId
        wallets {
          id
          balance
          walletCurrency
        }
      }
    }
  }
`;
